### ISSUE TITLE
FindEverythingX - Auction creator can hold funds hostage when not providing privateKey

### ISSUE THREAD

**Issue by sherlock-admin3** [sherlock-audit/2024-03-axis-finance-judging#207]
FindEverythingX

high

# Auction creator can hold funds hostage when not providing privateKey

## Summary
Auction creator can hold funds hostage when not providing privateKey

## Vulnerability Detail
Contract: EMPAM.sol

After bids have been provided, there are two scenarios that can happen:

a) Bids can be requested for refund via refundBid (only until the auction has concluded)

b) Bids can be claimed via claimBids (only once they are settled)

Consider the following scenario:

- The auction has concluded and is waiting to be settled
- Settlement can only happen once the status is decrypted:

[https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L754](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L754)

(auctionData[lotId_].status != Auction.Status.Decrypted)

- The state is only set to decrypted, once all bids have been decrypted:

[https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L520](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L520)

if (auctionData[lotId_].nextDecryptIndex == bidIds.length) {         	auctionData[lotId_].status = Auction.Status.Decrypted;     	}

- Bids can only be decrypted once the valid private key has been provided:

[https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L450](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L450)

if (         	auctionData[lotId_].status != Auction.Status.Created             	|| auctionData[lotId_].privateKey == 0     	) {         	revert Auction_WrongState(lotId_);     	}

The problem: Only the auction creator (seller) has the privateKey, the privateKey is needed for the “submitPrivateKey” function. If the seller decides to not submit the privateKey, all funds are permanently held hostage until the seller decides to submit the key. This can be done for blackmailing or any other purposes. 

The reason why that works is because the auction has been concluded (refund is impossible) and the auction is not in the decrypted state (claimBids is impossible). Therefore, users have no ability to get their funds back nor receive the desired base tokens.

The following is stated in EMPA.md:

“What: Need to be able to encrypt bids and other data with a key that no interested party controls until the auction ends. This is to prevent insider dealing or other bad behavior.
Solution: API and private database that creates EC keypairs and store the private key until an auction ends. If an auction is cancelled, the private key is never released.
Axis will provide a simple API that provides key management for EC keypairs generated on auction creation. The API will generate a new keypair and store the private key in a database. The public key is returned to the user and stored on the auction contract. Once the auction ends, the private key is released from the database and can be used to decrypt the bids. The API will also provide a convenience function for returning the next bids that need to be decrypted for a given auction. This will allow anyone to decrypt the bids and submit them to the contract for verification.”

Therefore, this means that users can use the API to create a public and private key, the private key is then stored and released. This would effectively render this attack invalid. 

The problem: Users can use their own key management logic (as long as it incorporates the same logic), which then means they can create an auction with a valid public key and they also have the valid private key. The problem here is that in this scenario, the private key is not stored in the database and released to the public once the auction is concluded. 
The crux is that the smart contract accepts any auction creation.

Within the private thread, I have asked the team if such a situation is possible and Oighty answered that this is possible. Obviously I know from the Sherlock docs that this is not considered as proof. However, it might be beneficial for the judge to acknowledge this as possible scenario.

“The goal is for the protocol to be permissionless and community-owned, which alters the security considerations”

## Impact
IMPACT:

a) Blackmailing of funds

## Code Snippet
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L754
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L520
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L450

## Tool used

Manual Review

## Recommendation

Consider implementing logic which allows refunds a certain time after the auction has concluded, if the private key was not yet provided

Duplicate of #174
